src/clj/puppetlabs/services/ca/certificate_authority_service.clj

```diff
@@ -55,22 +55,23 @@
                        events)
              (ca/retrieve-ca-crl! ca-crl-file host-crl-file)))))
       (assoc context :auth-handler auth-handler
-                     :watcher watcher)))
+                     :watcher watcher
+                     :ca-settings settings)))
 
-  (initialize-master-ssl!
-   [this master-settings certname]
-   (let [settings (ca/config->ca-settings (get-config))]
-     (ca/initialize-master-ssl! master-settings certname settings)))
+(initialize-master-ssl!
+  [this master-settings certname]
+  (let [context (tk-services/service-context this)]
+    (ca/initialize-master-ssl! master-settings certname (:ca-settings context))))
 
-  (retrieve-ca-cert!
-    [this localcacert]
-    (ca/retrieve-ca-cert! (get-in-config [:puppetserver :cacert])
-                          localcacert))
+(retrieve-ca-cert!
+  [this localcacert]
+  (let [context (tk-services/service-context this)]
+    (ca/retrieve-ca-cert! (:cacert (:ca-settings context)) localcacert)))
```

Comment:
This isn't big enough to really warrant going back and changing it, (but I think it's fun to talk about (: ). For future reference, using multiple, nested map accesses like this isn't terribly idiomatic. Alternatives are:
```clj
(get-in context [:ca-settings :cacert])
```
Which is especially good if you have a default you want to supply as the third argument
or
```clj
(-> context :ca-settings :cacert)
```

also, if you are assigning a variable and then only using it once to dig another value from it you can just set the variable to that value initially, eg:
```clj
(let [cacert (-> this tk-services/service-context :ca-settings :cacert)]
  (ca/retrieve-ca-cert! cacert localcacert))
```

Reply:
Oh, thank you for pointing this out! I'm still learning a lot of the nuances of clojure, so I really appreciate it :) I think it might be worth going back and changing, especially since I'm already going to fix those white space issues you mentioned above.